cmake/templates/pkgConfig.cmake.in

```diff
@@ -123,6 +123,25 @@ foreach(library ${libraries})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   elseif(${library} MATCHES "^-l")
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
+  elseif(${library} MATCHES "^-")
+    # This is a linker flag/option (like -pthread)
+    # There's no standard variable for these, so create an interface library to hold it
+    if(NOT @PROJECT_NAME@_NUM_DUMMY_TARGETS)
+        set(@PROJECT_NAME@_NUM_DUMMY_TARGETS 0)
+    endif()
+    math(EXPR @PROJECT_NAME@_NUM_DUMMY_TARGETS "${@PROJECT_NAME@_NUM_DUMMY_TARGETS}+1")
+    set(interface_target_name "catkin::@PROJECT_NAME@::wrapped-linker-option${@PROJECT_NAME@_NUM_DUMMY_TARGETS}")
+    add_library("${interface_target_name}" INTERFACE IMPORTED)
```

Comment:
While this target babe must be globally unique the variable set above containing the number might be scoped (if this file is included in a scope). So I could see this fail since multiple library targets can get the same name.

Can you update the logic to test for the existence of the target (`if(TARGET ...)`) instead?

Reply:
Increments until it finds a unique target name in 7834a30eff81ef1bd03b5446691583263b6227e8 (I'm still waiting for tests on this one to finish)